app/src/main/java/com/sopt/now/gs/feature/reserve/ReserveFragment.kt

```diff
@@ -1,15 +1,230 @@
 package com.sopt.now.gs.feature.reserve
 
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
+import androidx.viewpager2.widget.ViewPager2
 import com.sopt.now.gs.R
 import com.sopt.now.gs.core.base.BindingFragment
 import com.sopt.now.gs.databinding.FragmentReserveBinding
-import com.sopt.now.gs.feature.util.KeyStorage
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlin.math.ceil
 
 class ReserveFragment : BindingFragment<FragmentReserveBinding>(R.layout.fragment_reserve) {
+    lateinit var banner: Job
+    private var bannerPosition = 0
+    private val bannerItems = ArrayList<ReserveBannerPage>()
+    private val menuListItems = mutableListOf<ReserveMenuListItem>()
+    private val category1MenuItems = mutableListOf<ReserveMenuCategoryItem>()
+    private val category2MenuItems = mutableListOf<ReserveMenuCategory2Item>()
+    private val discountMenuViewModel by viewModels<ReserveDiscountViewModel>()
+
     override fun initView() {
-        val memberId = arguments?.getInt(KeyStorage.USER_ID) ?: -1
-        if (memberId != -1) {
-            binding.tvReserveExample.text = "홈에서 받아온 Member ID: $memberId"
+        setBannerItems()
+        initBannerAdapter()
+        setBannerPositionText()
+        updateBannerPosition()
+
+        setMenuListItems()
+        initMenuListAdapter()
+
+        initDiscountMenuAdapter()
+
+        setMenuCategory1Items()
+        initMenuCategory1Adapter()
+
+        setMenuCategory2Items()
+        initMenuCategory2Adapter()
+
+        moveToTop()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        scrollJobCreate()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        banner.cancel()
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        banner.cancel()
+    }
+
+    fun scrollJobCreate() {
+        banner = lifecycleScope.launch {
+            delay(1500)
+            binding.vpReserveBanner.setCurrentItem(++bannerPosition, true)
+        }
+    }
+
+    private fun setBannerItems(){
+        //임시 데이터
+        bannerItems.let {
+            it.apply {
+                add(ReserveBannerPage(R.drawable.img_reserve_banner1, "1"))
+                add(ReserveBannerPage(R.drawable.img_reserve_banner2, "2"))
+            }
+        }
```

Comment:
p1 : 아마 지워질 코드이지만 let을 null체크 용으로 사용한다면 run을 사용하는게 더 낫습니다! 또 scopㄷ가 중복될 필요도 없어보여요!
```suggestion
       bannerItems?.run {
                 add(ReserveBannerPage(R.drawable.img_reserve_banner1, "1"))
                 add(ReserveBannerPage(R.drawable.img_reserve_banner2, "2"))
        }

```